butane_core/src/migrations/fsmigrations.rs

```diff
@@ -289,6 +289,21 @@ impl FsMigrations {
         self.save_state(&state)?;
         Ok(())
     }
+    /// Determines if a migration has been detached.
+    pub fn has_detached_migration(&self) -> Result<bool> {
+        let migration_series = self.all_migrations()?;
+        for entry in std::fs::read_dir(self.root.clone())? {
+            let path = entry?.path();
+            let file_name = path.file_name().unwrap();
+            if file_name == "current" || file_name == "state.json" {
```

Comment:
Rather than checking for `state.json` consider checking `path.is_dir()`. Only directories indicate migrations. Files can't be migrations themselves.